### ISSUE TITLE
Azure SQL Database Deployment: How to use Active Directory - Integrated authentication

### ISSUE THREAD

**Issue by Garand-Gary** [Microsoft/azure-pipelines-tasks#9998]
## Required Information
*Type*: Question
**Enter Task Name**: SqlAzureDacpacDeployment

## Environment
- Server - Azure Pipelines
- Agent - Hosted VS2017

## Issue Description
I am curious as to how the "Active Directory - Integrated" authentication option works with the Azure SQL Database Deployment task.

I have managed to get "Active Directory - Password" authentication working which is a big step forward from using "SQL Server Authentication", but it still requires a username and password to be managed.

I assumed that "Active Directory - Integrated" would use the SPN behind the service connection selected in the "Azure Subscription" property for the task, but this doesn't seem to be the case unless I am doing something wrong.

**TL;DR:** *How can I use the "Active Directory - Integrated" Authentication Type in the SqlAzureDacpacDeployment task?*

**Comment by Ajay-MS** [comment]
@Garand-Gary 

_Active Directory - Integrated_ runs with the same credentials by which you have configured your agent. It doesn't use SPN behind the service connection.

**Comment by Garand-Gary** [comment]
@Ajay-MS thanks for getting back to me, very much appreciated.

OK, so I was misunderstanding what this was used for. I guess that when using a hosted agent this option is of no use?

**Comment by Ajay-MS** [comment]
@Garand-Gary 

Yes, For hosted agent this option is not useful. 

Closing this thread. Feel free to reopen in case of any question.

**Comment by davidobrien1985** [comment]
@Ajay-MS Hi Ajay, how can I use this task though to run on a hosted agent and use the pipeline's AAD identity to connect to Azure SQL?
We are trying to configure logins that map to AAD groups (based on this: https://techcommunity.microsoft.com/t5/Azure-Database-Support-Blog/Lesson-Learned-23-How-to-add-AAD-Security-Groups-to-Azure-SQL/ba-p/368866 ) but in order for this to run the connection must be done from an AAD identity.
We can connect to the DB via user name and password (tested) but then we get the error `Only connections established with Active Directory accounts can create other Active Directory users`.

Thanks,
David

**Comment by michc-msft** [comment]
@Garand-Gary @Ajay-MS What permissions do I need to configure (on both the DB and on the Azure portal) to set up the Active Directory - Password option on the SQL Dacpac task? I have been struggling with this for a bit now.

**Comment by lohithosur** [comment]
@Garand-Gary , @Ajay-MS How did you get Active Directory - Password working? Could you please point to relevant documentation? It would be very helpful.
